packages/frontend/app/routes/webhooks.tsx

```diff
@@ -0,0 +1,170 @@
+    nextPageSearchParams.set('after', webhooks.pageInfo.endCursor)
+    if (type.length > 0) nextPageSearchParams.set('type', type.join(','))
+    nextPageUrl = `/webhooks?${nextPageSearchParams.toString()}`
+  }
+
+  return json({ webhooks, previousPageUrl, nextPageUrl, type })
+}
+
+export default function WebhookEventsPage() {
+  const { webhooks, previousPageUrl, nextPageUrl, type } =
+    useLoaderData<typeof loader>()
+  const [, setSearchParams] = useSearchParams()
+  const navigate = useNavigate()
+
+  function setTypeFilterParams(selectedType: WebhookEventType): void {
+    const selected = type
+
+    if (!selected.includes(selectedType)) {
+      selected.push(selectedType)
+      setSearchParams(new URLSearchParams({ type: selected.join(',') }))
+      return
+    }
+
+    setSearchParams(() => {
+      const newParams = selected.filter((t) => t !== selectedType).join(',')
+      if (newParams.length === 0) {
+        return ''
+      }
+
+      return new URLSearchParams({
+        type: newParams
+      })
+    })
+  }
+
+  return (
+    <>
+      <div className='pt-4 flex flex-col space-y-8'>
+        <div className='flex flex-col rounded-md bg-offwhite px-6'>
+          <PageHeader>
+            <div className='flex-1'>
+              <h3 className='text-2xl leading-10'>Webhook Events</h3>
+            </div>
+          </PageHeader>
+          <div className='p-1'>
+            <h3 className='text-lg font-bold'>Filters</h3>
+            <div className='flex items-center'>
+              <PopoverFilter
+                label='Event type'
+                values={type.length > 0 ? type : ['all']}
+                options={[
+                  {
+                    name: 'All',
+                    value: 'all',
+                    action: () => {
+                      navigate(``)
+                    }
+                  },
+                  ...Object.values(WebhookEventType).map((value) => ({
+                    name:
+                      value.charAt(0).toUpperCase() +
+                      value.slice(1).replace(/[_.]/g, ' '),
+                    value: value,
+                    action: () => {
+                      setTypeFilterParams(value)
+                    }
+                  }))
+                ]}
+              />
+            </div>
+          </div>
+          <Table>
+            <Table.Head columns={['ID', 'Type', 'Data']} />
+            <Table.Body>
+              {webhooks.edges.length ? (
+                webhooks.edges.map((webhook) => (
+                  <Table.Row key={webhook.node.id}>
+                    <Table.Cell>{webhook.node.id}</Table.Cell>
+                    <Table.Cell>{webhook.node.type}</Table.Cell>
+                    <Table.Cell>
+                      <Button
+                        aria-label='view webhook data'
+                        state={{ data: webhook.node.data }}
+                        to={'/webhooks/data'}
```

Comment:
Related to the above suggestion. This way we can maintain the search parameters.
```suggestion
                        to={`/webhooks/data${
                          searchParams ? `?${searchParams}` : null
                        }`}
```